### PR TITLE
 Fix wrong type checking in Optimizer.add() 

### PR DESCRIPTION
**Pull request by vbkaisetsu**

### CODE REVIEW

File: tests/optimizer.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    def tearDown(self):
+        pass
+
+    def test_optimizer_add(self):
+        model = TestModel()
+        p = Parameter([5], I.Constant(0))
+        optimizer = O.Adam()
+        optimizer.set_weight_decay(1e-6)
+        optimizer.set_gradient_clipping(5)
+        optimizer.add(model)
```
Comment (odashi) [inline tests/optimizer.py]: Add assertions to check added objects.

Comment (vbkaisetsu) [reply tests/optimizer.py]: Fixed by testing gradient before and after calling `Optimizer.reset_gradients()`.

**Comment by vbkaisetsu** [comment]
Fixed.

File: tests/optimizer.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    def test_optimizer_add(self):
+        model = TestModel()
+        p = Parameter([1], I.Constant(0))
+        p.gradient = tF.raw_input([1], [1])
+        optimizer = O.Adam()
+        optimizer.set_weight_decay(1e-6)
+        optimizer.set_gradient_clipping(5)
+        optimizer.add(model)
+        optimizer.add(p)
+        self.assertEqual(p.gradient.to_list()[0], 1)
```
Comment (odashi) [inline tests/optimizer.py]: Right operand should be a list if possible.

**Comment by vbkaisetsu** [comment]
Fixed.

**Pull request merged**